dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -979,14 +980,64 @@ public static int runCmd(String[] cmd, File workingDirectory, File outputFile) t
+        long softTimeout = -1;
+        long startTime = System.currentTimeMillis();
+
+        // We need to ensure that the hard timeout is large enough to avoid future issues
+        if (hardTimeout >= 30000) {
+            softTimeout = hardTimeout - 10000; // Soft timeout is 10 seconds less than hard timeout
+            //softTimeout = 1000;   //ONLY FOR TESTING REMOVE THIS LINE WHEN COMPLETE
+        }
+
         Process p = pb.start();
-        int exitCode = p.waitFor();
+        int exitCode = -1;
+        if (softTimeout > -1) {
+            boolean exitStatus = false;
+            boolean softTimeOutFailed = false;
+            while (!exitStatus) {
+                try {
+                    exitCode = p.exitValue();
+                    exitStatus = true;
+                } catch (IllegalThreadStateException e) {
+                    if ((System.currentTimeMillis() - startTime > softTimeout) && !softTimeOutFailed) {
+                        // Parse through the MVN logs for potential networking issues
+                        if (outputFile.exists() && outputFile.canRead()) {
+                            try (Scanner s = new Scanner(outputFile)) {
+                                // Get the last two lines from the MVN log
+                                String lastLine = "";
+                                String secondLastLine = ""; 
+                                while (s.hasNextLine()) {
+                                    secondLastLine = lastLine;
+                                    lastLine = s.nextLine().toLowerCase();
+                                }
+                                // Check if the last or second line has the text "downloading" or "downloaded"
+                                // Throw custom timeout error rather then the one provided by the JUnitTask
+                                if ((lastLine.contains("downloaded") || lastLine.contains("downloading")) || (secondLastLine.contains("downloaded") || secondLastLine.contains("downloading"))) {       
+                                    String timeoutMsg = "Timeout occurred. FAT timeout set to: " + System.getProperty("fat.timeout") + 
+                                                    ". It appears there were some issues gathering dependencies. This may be due to network issues such as slow download speeds.";
+                                    Log.info(c, "runCmd", timeoutMsg);
+                                    throw new AssertionFailedError(timeoutMsg);
+                                }
+                                softTimeOutFailed = true;
+                            } catch (FileNotFoundException FileError) {
+                                // Do nothing as we can't look at the MVN log. This leads to hard timeout handled by JUnitTask.
+                                softTimeOutFailed = true;
+                            }
+                        }
+                    }
+                }
```

Comment:
Instead of this code, we can use `p.waitFor(long timeout, TimeUnit units)` which was introduced in Java 8. Now that we no longer have to support Java 7 we can use this new java 8 method. If `p.waitFor(long,TimeUnit)` returns `false` (i.e. the timeout was reached) then we can report a failure and IIRC the relevant logs will be automatically included